app-emulation/qemu/qemu-7.1.0.ebuild

```diff
@@ -821,6 +822,57 @@ src_install() {
 		# Install binfmt/qemu.conf.
 		insinto "/usr/share/qemu/binfmt.d"
 		doins "${T}/qemu.conf"
+
+		insinto "/lib/binfmt.d"
+		"${S}"/scripts/qemu-binfmt-conf.sh \
+			--systemd ALL \
+			--exportdir "${T}/binfmt.d" \
+			--qemu-path "${EPREFIX}/usr/bin" || die
+
+		# We don't want to install an interpreter for the native arch.
+		# Sorting order matches QEMU_USER_TARGETS, approximately
+		delete_binfmt_file() {
+			[[ -z ${1} ]] && die "Blank argument given to delete_binfmt_file!"
+
+			# Intentionally no die as they may not exist if not enabled in QEMU_USER_TARGETS
+			rm -v "${T}"/binfmt.d/qemu-${1}*
+		}
+
+		if use arm64 ; then
+			delete_binfmt_file aarch64
+		elif use alpha ; then
+			delete_binfmt_file alpha
+		elif use arm ; then
+			delete_binfmt_file arm
+		elif use hppa ; then
+			delete_binfmt_file hppa
+		elif use m68k ; then
+			delete_binfmt_file hppa
+		elif use ppc64 ; then
+			delete_binfmt_file ppc64
+		elif use ppc ; then
+			delete_binfmt_file ppc
+		elif use sparc ; then
+			delete_binfmt_file sparc
+		elif use mips ; then
+			delete_binfmt_file mips
+		elif use s390 ; then
+			delete_binfmt_file s390
+		elif use amd64 || use x86 ; then
+			delete_binfmt_file x86_64
+			delete_binfmt_file i386
+			delete_binfmt_file i586
+		elif use riscv ; then
```

Comment:
Thinking about it, we should probably die if we deleted 0 files because it implies we're missing a mapping?

Reply:
It's fine not to include one's native target in `QEMU_USER_TARGETS` so it's possible that we actually delete 0 files?

Reply:
Ah, right, good point.